bokeh/util/serialization.py

```diff
@@ -70,6 +68,26 @@
 # Globals and constants
 #-----------------------------------------------------------------------------
 
+_DATETIME_TYPES = set()
+
+def _compute_datetime_types() -> Set[type]:
+    if not _DATETIME_TYPES:
+        _DATETIME_TYPES.add(dt.time)
+        _DATETIME_TYPES.add(dt.datetime)
+        _DATETIME_TYPES.add(np.datetime64)
+        pd = import_optional('pandas')
+        if pd:
+            _DATETIME_TYPES.add(pd.Timestamp)
+            _DATETIME_TYPES.add(pd.Timedelta)
+            _DATETIME_TYPES.add(pd.Period)
+            _DATETIME_TYPES.add(type(pd.NaT))
+    return _DATETIME_TYPES
+
+def __getattr__(name: str) -> Any:
+    if name == "DATETIME_TYPES":
+        return _compute_datetime_types()
+    raise AttributeError
```

Comment:
I'm ambivalent about whether this is needed, especially landing on `branch-3.0` happy to remove this if anyone has an opinion.